### ISSUE TITLE
Training of severity encoder

### ISSUE THREAD

**Issue by jiayangshi** [z-fabian/flash-diffusion#1]
Hi,

thanks for the nice paper and code. I would like to ask if you could share more information about the training of severity encoder. You mentioned in the paper "Training the severity encoder is efficient, as it can be done by fine-tuning a pre-trained encoder". 

How do you map the latent variables encoded by the encode to severity levels? Do you use a single severity encoder for different tasks, or do you train for each task independently?

Thank you!

**Comment by z-fabian** [comment]
Hi @jiayangshi,

> How do you map the latent variables encoded by the encode to severity levels?

The severity encoder is trained to output both the latent estimate and the severity level (see Eq. 2 in the paper how for the particular loss we used). Given the severity, we find the optimal start time in the diffusion process using a notion of SNR matching (see Eq. 3 in the paper)

> Do you use a single severity encoder for different tasks, or do you train for each task independently?

We train a severity encoder separately for each task. You can find pretrained severity encoder checkpoints for various tasks [here](https://github.com/z-fabian/flash-diffusion?tab=readme-ov-file#pretrained-severity-encoders). We also provided a sample script how to train on your own task [here](https://github.com/z-fabian/flash-diffusion?tab=readme-ov-file#severity-encoder-training).

Hope this helps!
